app/controllers/profiles_controller.rb

```diff
@@ -12,6 +12,16 @@ def update
     end
   end
 
+  def search
+    @results = []
+    Profile.all.each do |profile|
+      if (@profile.distance_to(profile.postcode) <= params[:distance].to_f) && (profile != @profile)
+        @results << [profile, @profile.distance_to(profile.postcode).round(2)]
+      end
+    end 
+     @results.sort!{|a, b| a[1] <=> b[1]}
+  end
```

Comment:
This looks like it would work. But there's a fundamental issue here related to scalability. Here you're using `all` to find every result from the database `select * from profiles`. Then, we're allocating ActiveRecord models to accommodate them all, which involves a lot of memory. Then, we're doing the comparison in memory, and finally an in-memory sort. All of this would work for small numbers of profiles (into the 1000s) but once we get into higher numbers this isn't going to cut it. Instead, we need to delegate this work to the database, which is designed for this and is much faster. 

Take a look at what `Profile.near` does in pry or the docs, and use `order` to do your sorting. You'll find it's easier and faster!